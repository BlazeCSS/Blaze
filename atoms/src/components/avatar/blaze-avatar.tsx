import { Component, Prop } from '@stencil/core';

@Component({
  tag: 'blaze-avatar'
})
export class Avatar {

  @Prop() size: string = '';
  @Prop() src: string;
  @Prop() src2: string;
  @Prop() text: string;

  render() {
    if (!this.src && !this.text) return;

    const sizeClass = this.size.length > 0 ? `u-${this.size}` : '';

    return (
      <div class={`c-text c-avatar ${sizeClass}`} data-text={this.text}>
        {this.src && <img class="c-avatar__img" src={this.src} />}
        {this.src2 && <img class="c-avatar__img" src={this.src2} />}
      </div>
    );
  }
}
