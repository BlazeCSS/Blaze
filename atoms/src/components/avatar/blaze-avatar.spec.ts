import { flush, render } from '@stencil/core/testing';
import { Avatar } from './blaze-avatar';

describe('Avatar', () => {
  it('should build', () => {
    expect(new Avatar()).toBeTruthy();
  });

  describe('renders', () => {
    let element;

    const snapIt = (name, html) => {
      it(name, async () => {
        element = await render({
          components: [Avatar],
          html
        });
        await flush(element);

        expect(element).toMatchSnapshot();
      });
    };

    snapIt(
      'nothing if no attributes are set',
      '<blaze-avatar></blaze-avatar>'
    );

    snapIt(
      'text inside',
      '<blaze-avatar text="GP"></blaze-avatar>'
    );

    snapIt(
      'correctly sized circles',
      '<blaze-avatar size="super" text="GP"></blaze-avatar>'
    );

    snapIt(
      'an image',
      '<blaze-avatar src="my-test-image.png"></blaze-avatar>'
    );

    snapIt(
      '2 images',
      '<blaze-avatar src="my-test-image.png" src2="my-second-test-image.png"></blaze-avatar>'
    );

    snapIt(
      'images, text and sizes it correctly',
      '<blaze-avatar src="my-test-image.png" src2="my-second-test-image.png" size="super" text="GP"></blaze-avatar>'
    );
  });
});
