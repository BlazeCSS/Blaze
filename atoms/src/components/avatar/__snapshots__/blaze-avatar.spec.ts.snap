// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders 2 images 1`] = `
<blaze-avatar
  class="hydrated"
  src="my-test-image.png"
  src2="my-second-test-image.png"
>
  <!---->
  <div
    class="c-text c-avatar"
  >
    <img
      class="c-avatar__img"
      src="my-test-image.png"
    />
    <img
      class="c-avatar__img"
      src="my-second-test-image.png"
    />
  </div>
</blaze-avatar>
`;

exports[`Avatar renders an image 1`] = `
<blaze-avatar
  class="hydrated"
  src="my-test-image.png"
>
  <!---->
  <div
    class="c-text c-avatar"
  >
    <img
      class="c-avatar__img"
      src="my-test-image.png"
    />
    
  </div>
</blaze-avatar>
`;

exports[`Avatar renders correctly sized circles 1`] = `
<blaze-avatar
  class="hydrated"
  size="super"
  text="GP"
>
  <!---->
  <div
    class="c-text c-avatar u-super"
    data-text="GP"
  >
    
  </div>
</blaze-avatar>
`;

exports[`Avatar renders images, text and sizes it correctly 1`] = `
<blaze-avatar
  class="hydrated"
  size="super"
  src="my-test-image.png"
  src2="my-second-test-image.png"
  text="GP"
>
  <!---->
  <div
    class="c-text c-avatar u-super"
    data-text="GP"
  >
    <img
      class="c-avatar__img"
      src="my-test-image.png"
    />
    <img
      class="c-avatar__img"
      src="my-second-test-image.png"
    />
  </div>
</blaze-avatar>
`;

exports[`Avatar renders nothing if no attributes are set 1`] = `
<blaze-avatar
  class="hydrated"
>
  <!---->
  
</blaze-avatar>
`;

exports[`Avatar renders text inside 1`] = `
<blaze-avatar
  class="hydrated"
  text="GP"
>
  <!---->
  <div
    class="c-text c-avatar"
    data-text="GP"
  >
    
  </div>
</blaze-avatar>
`;
